src/provinces/province.cpp

```diff
@@ -716,6 +717,51 @@ void change_province_owner(sys::state& state, dcon::province_id id, dcon::nation
 			if(state.world.province_get_building_level(id, uint8_t(economy::province_building_type::naval_base)) > 0)
 				state.world.state_instance_set_naval_base_is_taken(new_si, true);
 
+			auto new_market = state.world.create_market();
+
+			// set prices to something to avoid infinite demand:
+
+			state.world.for_each_commodity([&](auto cid){
+				state.world.market_set_price(new_market, cid, 10.f);
+			});
```

Comment:
Would it make sense to copy the market data from the previous state's market? A common situation will be a whole state (and hence its market) changing owners, but this is implemented as a sequence of province ownership changes. Of course, a single province could also split up the state, so I guess only copy the data if it wouldn't break things if the old market kept existing.